### PR TITLE
Initial Thoughts on Usecase plans

### PR DESCRIPTION
**Pull request by sandersms**

### CODE REVIEW

File: USECASES.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Use Cases
+
+## Usage Deployments - Capabilities desired by end users
+
+- Virtual Switch Offload
+- Network Service Offload
+- NVMeoF
+- Storage Offload
```
Comment (glimchb) [inline USECASES.md:8]: not sure what `Storage Offload` is... storage services?

File: USECASES.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## Phases of Operation - API considerations
+
+- Pre-provisioning
+- Day 0 (Initial Startup)
+  - Discovery
+  - Initial Provisioning
+- Day N
+  - Re-provisioning
+  - Firmware Update
+  - OS Update
```
Comment (glimchb) [inline USECASES.md:21]: I would add "retirement"...

File: USECASES.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+## Interfaces
+
+- Configuration Interface
+  - Network facing (Ethernet inband or out-of-band)
+  - Host facing (Network port over PCIe)
```
Comment (glimchb) [inline USECASES.md:27]: I would add "BMC-facing"

File: USECASES.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+- Consumption Interface
+  - Virtual Function
+    - Network VF Interface
+    - Virtual Storage (VF Interface)
```
Comment (glimchb) [inline USECASES.md:31]: -AI/ML?
-Security?

File: USECASES.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+- VF/PF
+- Access Security
+
+## UseCases
+
+### Network Services Offload
+
+The Network Services Offload use case will build a foundation for the examples that can be put together for the OPI.  The basic topology is shown in the diagram.  It consists of two servers, with D/IPUs that are connected through a network switch.
+
+![Network Services Offload Use Case](doc/minutes/images/API-Use-Case-1.png)
```
Comment (glimchb) [inline USECASES.md:45]: we can create similar diagram for storage. if we rename server 1 to Network Storage Array (or EBOF)

File: USECASES.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    - Network VF Interface
+    - Virtual Storage (VF Interface)
+
+## Building Blocks
+
+- gRPC
+- VF/PF
+- Access Security
+
+## UseCases
```
Comment (pdp2shirts) [inline USECASES.md:39]: should this be Use Cases. title case rather than camel case

**Review by pdp2shirts — COMMENTED** [review]
left a comment @ line 39

File: USECASES.md (hunk 6 above)
Comment (sandersms) [reply USECASES.md:39]: Fixing in the next update to read Use Case Examples

File: USECASES.md (hunk 1 above)
Comment (sandersms) [reply USECASES.md:8]: Updated to include the services.

File: USECASES.md (hunk 2 above)
Comment (sandersms) [reply USECASES.md:21]: Done - in the next update

File: USECASES.md (hunk 3 above)
Comment (sandersms) [reply USECASES.md:27]: In the next update

File: USECASES.md (hunk 4 above)
Comment (sandersms) [reply USECASES.md:31]: I added some questions to ignite the thought around these two topics

File: USECASES.md (hunk 5 above)
Comment (sandersms) [reply USECASES.md:45]: We should add additional diagrams and discussions around Storage and Firewall to align with the various use cases from the use case workstream.

Comment (glimchb) [reply USECASES.md:45]: I added one

**Pull request merged**